nuclear/module/NUClearModule.cmake

```diff
@@ -191,6 +191,9 @@ function(NUCLEAR_MODULE)
   # Put it in an IDE group for the module's directory
   set_target_properties(${module_target_name} PROPERTIES FOLDER ${module_path})
 
+  # Set the C++ standard
+  target_compile_features(${module_target_name} PUBLIC cxx_std_20)
```

Comment:
I wouldn't add this here either, not until you're using c++20 almost everywhere. Until then you can add it on a per-module basis, or rely on it coming in through utilities that use it

Reply:
So I was trying to compile the codebase with clang so we can get clang sanitizers, but clang complains because apparently I introduced some C++20isms in #516 with `constexpr` usage. GCC doesn't seem to care, but clang gets quite upset. 
I'll set it on a per-module basis for those then :+1: